File: mm.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

public class hmwk04
{
   public static void Main(string[] args)
   {

	   // skip first line of file because its just headers
		StreamReader input = new StreamReader(args[0]);
		input.ReadLine();
		
		//go through each line
		while (!input.EndOfStream) {
			// split each line by tabs
			string student = input.ReadLine();
			string[] fields = student.Split('\t');

			//delete file if exists; mainly used for testing
			if(File.Exists(fields[1])){
			    File.Delete(fields[1]);
			}

			// copy template file, then make SB of template file to replace 
			// keywords in file with
			File.Copy(args[1],fields[1]);
			string studentFileStr = File.ReadAllText(args[1]); 
			StringBuilder studentFileBldr = new StringBuilder(studentFileStr);
			
			studentFileBldr.Replace("<<NAME>>", fields[0]);
			studentFileBldr.Replace("<<COURSE>>", fields[2]);
			studentFileBldr.Replace("<<ID>>", fields[1]);
			studentFileBldr.Replace("<<TOTAL>>", fields[12]);
			studentFileBldr.Replace("<<SUBTOTAL>>", fields[11]);
			studentFileBldr.Replace("<<LATEDEDUCTION>>", fields[6]);
			studentFileBldr.Replace("<<DUE>>", fields[3]);
			studentFileBldr.Replace("<<SUBMITTED>>", fields[4]);
			studentFileBldr.Replace("<<MINUTESLATE>>", fields[5]);
			studentFileBldr.Replace("<<P1>>", fields[7]);
			studentFileBldr.Replace("<<P1COMMENTS>>", fields[8]);
			studentFileBldr.Replace("<<P2>>", fields[9]);
			studentFileBldr.Replace("<<P2COMMENTS>>", fields[10]);
		
			//dump SB
			File.WriteAllText(fields[1], studentFileBldr.ToString());
		}
	input.Close();
		
	}	
}

File: mm.py
import re
import fileinput
import shutil
from sys import argv

script, studentsFile, templateFile = argv

#  was going to use this class until I realized it wasnt needed. Saving it for future reference.

class Student:
    name = "";
    ID = "";
    course = "";
    due	= "";
    submitted = "";
    minutesLate	= "";
    lateDeduction = "";
    p1 = "";
    P1comments = "";
    P2 = "";
    P2comments = "";
    subtotal = "";
    total= "";

    def __init__(self, name, ID, course, due, submitted, minutesLate, lateDedction, p1, p1comments, p2, p2comments, subtotal, total):
        self.name = name
        self.ID = ID 
        self.course = course 
        self.due = due
        self.submitted = submitted 
        self.minutesLate = minutesLate 
        self.lateDedction 
        self.p1 = p1 
        self.p1comments = p1comments 
        self.p2 = p2 
        p2comments= p2comments 
        self.subtotal = subtotal 
        self.total = total 

# go through lines of file, strip by new line, then by tabs
lines = [line.rstrip('\n') for line in open(studentsFile)] 
#students is in format: SplitLines[SplitTabs[]]
students = []

for i in range(len(lines)):
    if i != 0:
        students.append(re.split(r'\t+', lines[i].rstrip('\t')))

# take in a list (called student) of student's attributes 
# creates a file named by the student's ID filled out by the template file
def createStudentFile(student):
    newFileName = student[1] + '.py'
    shutil.copyfile(templateFile, newFileName)
    #with open(newFileName) as f:
        #for c in f.read():
    lineNum = 0
    
    #get number of lines in file
    numLines = sum(1 for _ in open(templateFile))
    # check for keywords to replace and replace them if found
    with fileinput.FileInput(newFileName, inplace=True, backup='') as file:
        for line in file:
            if lineNum == 0:
                lineZero = line.replace("<<NAME>>", student[0])
                print(lineZero.replace("<<COURSE>>", student[2]), end='')
            elif lineNum == 1:
                print(line.replace("<<ID>>", student[1]), end='')
            elif lineNum == 4:
                print(line.replace("<<DUE>>", student[3]), end='')
            elif lineNum == 5:
                print(line.replace("<<SUBMITTED>>", student[4]), end='')
            elif lineNum == 6:
                print(line.replace("<<MINUTESLATE>>", student[5]), end='')
            elif lineNum == 7:
                print(lineTwo.replace("<<LATEDEDUCTION>>", student[6]), end='')
            elif lineNum == 9:
                print(line.replace("<<P1>>", student[7]), end='')
            elif lineNum == 10: 
                print(line.replace("<<P1COMMENTS>>", student[8]), end='')
            elif lineNum == 12:
                print(line.replace("<<P2>>", student[9]), end='')
            elif lineNum == 13:
                print(line.replace("<<P2COMMENTS>>", student[10]), end='')
            elif lineNum == 2:
                lineTwo = line.replace("<<TOTAL>>", student[12])
                lineTwo = lineTwo.replace("<<SUBTOTAL>>", student[11])
                print(lineTwo.replace("<<LATEDEDUCTION>>", student[6]), end='')
            lineNum += 1

# create files for each student
for i in range(len(students)):
        createStudentFile(students[i])


